packages/thirdweb/src/transaction/transaction-store.ts

```diff
@@ -0,0 +1,106 @@
+import type { Chain } from "../chains/types.js";
+import type { ThirdwebClient } from "../client/client.js";
+import type { Hex } from "../utils/encoding/hex.js";
+import { stringify } from "../utils/json.js";
+import { LocalStorage } from "../wallets/in-app/web/utils/Storage/LocalStorage.js";
+import type { Account } from "../wallets/interfaces/wallet.js";
+import { waitForReceipt } from "./actions/wait-for-tx-receipt.js";
+import type { TransactionReceipt } from "./types.js";
+
+type StoredTransaction = {
+  transactionHash: Hex;
+  status: "PENDING" | "CONFIRMED";
+  receipt?: TransactionReceipt;
+};
+
+let storage: LocalStorage | undefined;
```

Comment:
This should be asyncStorage ideally. Should work for both web and RN

Reply:
we prob need to pass the storage in each of the functions below so it can be injected from the top

Reply:
yeah this is actually a blocker.

Reply:
@joaquim-verges
Had a talk with Jonas about this, we're just gonna store in memory for now